app/src/main/java/com/rage/clamber/Fragments/HomeActivity/Walls/ClimbDetailFragment.java

```diff
@@ -304,20 +316,31 @@ public void onFailure(Call<Double> call, Throwable t) {
 
     /**
      * Method to add a rating to the database for a user based off of up and down arrow button clicks.
+     *
      * @param request - request contains the climb id, user name and rating. Information is populated
      *                in the onClick methods for the up and down arrows.
      */
-    public void addRatingForUser(UserRatingRequest request){
+    public void addRatingForUser(UserRatingRequest request) {
         final Call<Integer> addRatingCall = ApiManager.getClamberService().addUserRating(request);
         addRatingCall.enqueue(new Callback<Integer>() {
             @Override
             public void onResponse(Call<Integer> call, Response<Integer> response) {
                 if (response.code() == 200) {
-                    yourRatingTextView.setText(getContext().getString(R.string.your_rating_s, response.body()));
+                    if (response.body() == -1){
```

Comment:
This is essentially the same code as above for the userRating text, go ahead and make a function that converts a raw double into an avg rating string, and a function that converts an int into an appropriate rating string. Death to code duplication!